libqtile/scripts/migrate.py

```diff
@@ -115,6 +119,54 @@ def new_at_current_to_new_client_position(config):
     )
 
 
+def textbox_widget_padding():
+    """
+    Adding padding mixin to _TextBox means configs need to
+    be updated. "padding" should just impact left/right padding
+    so needs to be replaced with "padding_x" to allow vertical
+    padding to be adjusted independently.
+    """
+    def update_padding(ln, cap, fn):
+        for c in cap.get("class_arguments"):
+            for leaf in c.leaves():
+                if leaf.value == "padding":
+                    leaf.value = "padding_x"
+
+    migrations = []
+
+    textwidgets = [getattr(libqtile.widget, w) for w in libqtile.widget.__all__]
+    textwidgets = [widget.__name__ for widget in textwidgets if inspect.isclass(widget) and issubclass(widget, _TextBox)]
```

Comment:
Not convinced this is best approach. As noted in comments, this will skip widgets that have third party dependencies.

Is it better to have hard-coded list of `_TextBox` widgets?

Reply:
It'll only skip things that have third party dependencies that aren't currently installed, right? I think that's fine - people won't have widgets in their configs that they don't have dependencies for.

Reply:
D'oh! Of course.